src/zenml/orchestrators/base_orchestrator.py

```diff
@@ -376,6 +376,7 @@ def run_step(self, step: "Step") -> Optional[data_types.ExecutionInfo]:
                 execution_info = self._execute_step(component_launcher)
             except:  # noqa: E722
                 self._publish_failed_run(run_name_or_id=run_model.id)
+                stack.cleanup_step_run(info=step_run_info, step_failed=True)
```

Comment:
this try-except logic might change a bit after #1103, @schustmi will the general flow of calling `stack.cleanup_step_run(info=step_run_info, step_failed=True)` after failures still work with our changes?

Reply:
I think that should still work as expected yes

Reply:
Ah I just saw, in general it will work but the current implementation has a flaw:
In case the step execution fails, it would call `cleanup_step_run` twice, first with `step_failed=True` followed by `step_failed=False` (due to the `finally` clause which always gets executed.

Two potentials solutions:

```python
step_failed = False
try:
  ...
except:
  self._publish_failed_run(run_name_or_id=run_model.id)
  step_failed = True
finally:
  self.cleanup_step_run(info=step_run_info, step_failed=step_failed)
```

```python
try:
  ...
except:
  self._publish_failed_run(run_name_or_id=run_model.id)
  self.cleanup_step_run(info=step_run_info, step_failed=True)
else:
  self.cleanup_step_run(info=step_run_info, step_failed=False)
```

Reply:
@schustmi what about including this in `StepRunInfo` as @htahir1 suggested or is `StepRunInfo` only for the information to run the step not the state of the run?